platforms/core-execution/build-cache-spi/src/main/java/org/gradle/caching/BuildCacheException.java

```diff
@@ -16,14 +16,12 @@
 
 package org.gradle.caching;
 
-import org.gradle.api.GradleException;
-
 /**
  * <p><code>BuildCacheException</code> is the base class of all exceptions thrown by a {@link BuildCacheService}.</p>
  *
  * @since 3.3
  */
-public class BuildCacheException extends GradleException {
+public class BuildCacheException extends RuntimeException {
```

Comment:
🤔 Is that a breaking change?

Reply:
I don't think so. These exceptions don't make it out of Gradle's internals to a place where user code could catch them. The only place the user might encounter them is if they created their own `BuildCacheService`, but there they would be throwing the exception, where its ancestry doesn't matter.

Some third-party code out there that explicitly casts a `BuildCacheException` as `GradleException` cannot be ruled out, but I think it's very-very unlikely to exist.